lib/pure/batteries.nim

```diff
@@ -0,0 +1,57 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2012 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+when defined(nimdoc) and isMainModule:
+  from std/compileSettings import nil
+  when compileSettings.querySetting(compileSettings.SingleValueSetting.projectFull) == currentSourcePath:
+    ## This is a module that simply imports common modules for your convenience:
```

Comment:
> thanks for your work on this, I have added some minor remarks.
> 
> A few more things:
> 
> * it might be worth putting also a reference to this in the documentation of standard library (maybe in the first section?): https://nim-lang.org/docs/lib.html

This would be done by editing doc/lib.md?

> * I have looked again at the list of stdlib modules to see if there were other candidates for inclusion in batteries. I think `enumerate` should also probably added (why is it not in sugar?) and probably it might make sense to add also `setutils` (I did not now about it but at least `toSet` seems a nice inclusion). I did [previously](https://github.com/nim-lang/Nim/pull/20022#issuecomment-1185142892) also suggested `bitops`, `unicode`, `json` as possible candidate for inclusion but I agree with the choice made here. I am still on the fence on whether `with` should made or not the list, maybe so (I would probably fine with it if it were in macros, again not sure why this is not the case).

I also reviewed the list and I basically considered the same ones you mentioned in addition to Complex. In fact I had the same questions about enumerate.nim and with.nim than you did. The one I was less sure about was json because working with json is perhaps not as common and because there are plenty of alternatives like jsony that are very popular. In any case in the end I decided to be conservative since I guess it is easier to add additional modules later than removing them once they are in. I can add more now if you feel strongly about some of them but maybe it would be better to first get this first version in and then add more on a later PR (perhaps after having some discussion in the forum)?

> * as @Yardanico [mentioned](https://github.com/nim-lang/Nim/pull/20022#issuecomment-1183675989) it might be worth adding a changelog line for this

That would be done on the "Standard library additions and changes" section of changelog.md?

> * should we get additional eyes (e.g. @dom96, ...) or public attention (maybe even a forum post) on the selection of modules?

I think that is a good idea, but as I said I'd prefer to get this PR in and then add more after a discussion in the forum. Is that ok?

Reply:
> This would be done by editing doc/lib.md?

yes, it should

> I can add more now if you feel strongly about some of them but maybe it would be better to first get this first version in and then add more on a later PR (perhaps after having some discussion in the forum)?

nope, no strong feelings. fine to add later. and I guess it is probably not me who should have last words on this PR, just trying to help out :)

> That would be done on the "Standard library additions and changes" section of changelog.md?

I think so, yes

> as I said I'd prefer to get this PR in and then add more after a discussion in the forum. Is that ok?

fine for me